sdk/communication/azure-communication-messages/azure/communication/messages/_shared/user_credential.py

```diff
@@ -6,39 +6,115 @@
      The returned token must be valid (expiration date must be in the future).
     :paramtype token_refresher: Callable[[], AccessToken]
     :keyword bool proactive_refresh: Whether to refresh the token proactively or not.
      If the proactive refreshing is enabled ('proactive_refresh' is true), the credential will use
-     a background thread to attempt to refresh the token within 10 minutes before the cached token expires,
-     the proactive refresh will request a new token by calling the 'token_refresher' callback.
+     a background thread to attempt to refresh the token within 10 minutes before the cached token expires.
+     The proactive refresh will request a new token by calling the 'token_refresher' callback.
      When 'proactive_refresh' is enabled, the Credential object must be either run within a context manager
      or the 'close' method must be called once the object usage has been finished.
-    :raises: TypeError if paramater 'token' is not a string
+    :keyword str resource_endpoint: The endpoint URL of the resource to authenticate against.
+    :keyword token_credential: The credential to use for token exchange.
+    :paramtype token_credential: ~azure.core.credentials.TokenCredential
+    :keyword list[str] scopes: The scopes to request during the token exchange. If not provided,
+     a default value will be used: https://communication.azure.com/clients/.default
+
+    :raises: TypeError if parameter 'token' is not a string
     :raises: ValueError if the 'proactive_refresh' is enabled without providing the 'token_refresher' callable.
     """
 
     _ON_DEMAND_REFRESHING_INTERVAL_MINUTES = 2
     _DEFAULT_AUTOREFRESH_INTERVAL_MINUTES = 10
 
-    def __init__(self, token: str, **kwargs: Any):
-        if not isinstance(token, str):
-            raise TypeError("Token must be a string.")
-        self._token = create_access_token(token)
-        self._token_refresher = kwargs.pop("token_refresher", None)
-        self._proactive_refresh = kwargs.pop("proactive_refresh", False)
-        if self._proactive_refresh and self._token_refresher is None:
-            raise ValueError("When 'proactive_refresh' is True, 'token_refresher' must not be None.")
+    @overload
+    def __init__(
+            self,
+            token: str,
+            *,
+            token_refresher: Optional[Callable[[], AccessToken]] = None,
+            proactive_refresh: bool = False,
+            **kwargs: Any
+    ):
+        """
+        Initializes the CommunicationTokenCredential.
+
+        :param str token: The token used to authenticate to an Azure Communication service.
+        :param token_refresher: Optional callable to refresh the token.
+        :param proactive_refresh: Whether to refresh the token proactively.
+        :param kwargs: Additional keyword arguments.
+        """
+        # pylint: disable=W0233
+        self.__init__(token, token_refresher=token_refresher, proactive_refresh=proactive_refresh, **kwargs)
```

Comment:
Remove this line of code from the overload - there should not be any implementation in a method marked with @overload. This is defined purely for static typing and won't ever actually be called.

Reply:
Please do this for all the overloads defined in the user_credential/user_credential_async.py. This should clear up the mypy errors you're seeing.

Reply:
done